src/storage/single_file_block_manager.cpp

```diff
@@ -432,9 +436,8 @@ void SingleFileBlockManager::Truncate() {
 		return;
 	}
 	// truncate the file
-	for (idx_t i = 0; i < blocks_to_truncate; i++) {
-		free_list.erase(max_block + i);
-	}
+	free_list.erase(free_list.lower_bound(max_block), free_list.end());
+	newly_freed_list.erase(newly_freed_list.lower_bound(max_block), newly_freed_list.end());
```

Comment:
I was not obvious to me looking here that both free_list and newly_freed_list are guaranteed to be sorted (given they are backed by a set<>).

Unsure if that should be made more visible.

Reply:
It's part of the spec that std::set is sorted. Are you suggesting that we rename these members, to make it clearer that they're sorted?